### ISSUE TITLE
required_with with custom struct

### ISSUE THREAD

**Issue by li601972633** [go-playground/validator#871]
- [x] I have looked at the documentation [here](https://pkg.go.dev/github.com/go-playground/validator/v10#section-documentation) first?
- [x] I have looked at the examples provided that may showcase my question [here](/_examples)?
### Package version eg. v9, v10: 
v10

### Issue, Question or Enhancement:

How can I use `required_with` tag on my custom struct.
For example, like the structure shown below, when UseA is true, the validator validate the `required_with` tag on filed TestA, and when UseA is false, skip the validating on filed TestA.

### Code sample, to showcase or reproduce:

```go

type A struct {
	Size int `validate:"gt=1"`
	Num int `validate:"gt=1"`
}

type Test struct {
	TestA A `validate:"required_with=ByPage"`
	UseA bool
}
```

**Comment by zemzale** [comment]
I don't think this is supported right now with struct tags. You can get this however by creating your own custom validator function. See https://pkg.go.dev/github.com/go-playground/validator/v10#hdr-Custom_Validation_Functions

Probably can be marked as a duplicate of #491
